src/collection-preferences/content-display/index.tsx

```diff
@@ -0,0 +1,121 @@
+  value?: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>;
+}
+
+export default function ContentDisplayPreference({
+  title,
+  description,
+  options,
+  value = options.map(({ id }) => ({
+    id,
+    visible: true,
+  })),
+  onChange,
+  liveAnnouncementDndStarted,
+  liveAnnouncementDndItemReordered,
+  liveAnnouncementDndItemCommitted,
+  liveAnnouncementDndDiscarded,
+  dragHandleAriaDescription,
+  dragHandleAriaLabel,
+}: ContentDisplayPreferenceProps) {
+  const idPrefix = useUniqueId(componentPrefix);
+
+  const isVisible = (id: string) => {
+    const option = options.find(option => option.id === id);
+    return !!(option?.alwaysVisible || value.find(item => item.id === id)?.visible);
+  };
+
+  const onToggle = (id: string) => {
+    onChange(value.map(item => (item.id === id ? { ...item, visible: !isVisible(id) } : item)));
+  };
+
+  const titleId = `${idPrefix}-title`;
+  const descriptionId = `${idPrefix}-description`;
+
+  const sortedOptions = getSortedOptions({ options, order: value });
+
+  const { activeItem, collisionDetection, handleKeyDown, isKeyboard, sensors, setActiveItem } = useDragAndDropReorder({
+    sortedOptions,
+  });
+
+  const announcements = useLiveAnnouncements({
+    isDragging: activeItem !== null,
+    liveAnnouncementDndStarted,
+    liveAnnouncementDndItemReordered,
+    liveAnnouncementDndItemCommitted,
+    liveAnnouncementDndDiscarded,
+    sortedOptions: value,
+  });
+
+  return (
+    <div className={styles[componentPrefix]}>
+      <h3 {...className('title')} id={titleId}>
+        {title}
+      </h3>
+      <p {...className('description')} id={descriptionId}>
+        {description}
+      </p>
+      <DndContext
+        sensors={sensors}
+        collisionDetection={collisionDetection}
+        accessibility={{
+          announcements,
+          restoreFocus: false,
+          screenReaderInstructions: dragHandleAriaDescription ? { draggable: dragHandleAriaDescription } : undefined,
+        }}
+        onDragStart={({ active }) => setActiveItem(active.id)}
+        onDragEnd={event => {
+          setActiveItem(null);
+          const { active, over } = event;
+
+          if (over && active.id !== over.id) {
+            const oldIndex = value.findIndex(({ id }) => id === active.id);
+            const newIndex = value.findIndex(({ id }) => id === over.id);
+            onChange(arrayMove([...value], oldIndex, newIndex));
+          }
+        }}
+        onDragCancel={() => setActiveItem(null)}
+      >
+        <ul {...className('option-list')} aria-describedby={descriptionId} aria-labelledby={titleId}>
+          <SortableContext items={sortedOptions.map(({ id }) => id)} strategy={verticalListSortingStrategy}>
+            {sortedOptions.map(option => {
+              return (
+                <SortableItem
+                  dragHandleAriaLabel={dragHandleAriaLabel}
+                  key={option.id}
+                  idPrefix={idPrefix}
```

Comment:
Done. I wanted to be extra safe about components eventually re-rendering and creating different IDs, but since we are using the `key` prop, this shouldn't happen.